weakmap.c

```diff
@@ -398,7 +432,33 @@ wmap_aset_replace(st_data_t *key, st_data_t *val, st_data_t new_key_ptr, int exi
     return ST_CONTINUE;
 }
 
-/* Creates a weak reference from the given key to the given value */
+/*
+ *  call-seq:
+ *    map[key] = value -> value
+ *
+ *  Associates the given +value+ with the given +key+; returns +value+.
```

Comment:
The method itself does not return `value`, it returns an integer.  The expression in the call-seq does, but only because assignment expressions return the right hand side.

Reply:
Well, in this case, I just copied that from `WeakKeyMap` docs, and that was apparently just copied from the [Hash docs](https://docs.ruby-lang.org/en/master/Hash.html#method-i-5B-5D-3D) :shrug: So I didn't put much thought in it.

But I agree it is factually incorrect.